In [1]:
## Import packages 

import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import statistics
import seaborn as sns
%matplotlib inline

## Load File

df = pd.read_csv('https://raw.githubusercontent.com/borja876/Thinkful-DataScience-Borja/master/WELLCOME_APCspend2013_forThinkful.csv', encoding='latin1')

In [2]:
## View DataFrame
#df

## Determine the five most common journals and the total articles for each

In [3]:
## View counts of each Journal Entry (as typed)
#journal_counts=df['Journal title'].value_counts()
#print(journal_counts)

In [4]:
## Create a function to count values of new DataFrames

def count(x): 
    return x.value_counts()

In [5]:
### Go through Each Value Count and standardize entries clearly referring to the same journal

In [6]:
## Standardize PLOS ONE entries

## Converting whole DataFrame to lowercase in the process
df_lower = df.applymap(lambda x: x.lower() if type(x) == str else x )

In [7]:
## View DataFrame after this change. all words are lowercase
# df_lower

## View Journal Title Counts after this change. 
count(df_lower['Journal title'])

plos one                                                                            190
journal of biological chemistry                                                      53
neuroimage                                                                           29
plos genetics                                                                        24
plos pathogens                                                                       24
nucleic acids research                                                               23
proceedings of the national academy of sciences                                      20
plos neglected tropical diseases                                                     20
human molecular genetics                                                             19
nature communications                                                                19
bmc public health                                                                    14
brain                           

**The five most common jourals (and the respective number of articles) are: PLOS One (190), Journal of Biological Chemistry (53), Neuroimage (29), PLOS Pathogens (24), and PLOS Genetics (24).**

## Next, calculate the mean, median, and standard deviation of the open-access cost per article for each journal

In [8]:
### Cleaning Article Cost Entries 

In [9]:
## removed currency symbols from cost column

## removed £
df_lower_currency1_removed = df_lower.apply(lambda x: x.str.replace('£', ''))

## removed $
df_lower_currency2_removed = df_lower_currency1_removed.apply(lambda x: x.str.replace('$', ''))

In [10]:
## remove extreme values

## Identify extreme values

#df_lower_currency2_removed['COST (£) charged to Wellcome (inc VAT when charged)'].apply(pd.to_numeric).describe()
#df_lower_currency2_removed['COST (£) charged to Wellcome (inc VAT when charged)'].apply(pd.to_numeric).value_counts()

##Identified 999999.00 as an extreme

In [11]:
## Find mean if extremes were removed

## Create Throwaway copy of main DataFrame
df_lower_currency2b_removed = df_lower_currency1_removed.apply(lambda x: x.str.replace('$', ''))

## Remove Extreme Values from temporary DataFrame
q = df_lower_currency2b_removed['COST (£) charged to Wellcome (inc VAT when charged)'].apply(pd.to_numeric).quantile(0.99)
df_lower_currency2b_removed = df_lower_currency2b_removed[df_lower_currency2b_removed['COST (£) charged to Wellcome (inc VAT when charged)'].apply(pd.to_numeric) < q]

## Get Statistics for temporary DataFrame
df_lower_currency2b_removed['COST (£) charged to Wellcome (inc VAT when charged)'].apply(pd.to_numeric).describe()

count      2080.000000
mean       2015.038038
std        6100.422956
min           0.000000
25%        1268.700000
50%        1852.505000
75%        2302.930000
max      201024.000000
Name: COST (£) charged to Wellcome (inc VAT when charged), dtype: float64

In [12]:
## Replace 999999.00 value with mean

df_lower_currency3_removed = df_lower_currency2_removed.apply(lambda x: x.str.replace('999999.00', '2015.04'))

In [13]:
## Check New DataFrame
#df_lower_currency3_removed['COST (£) charged to Wellcome (inc VAT when charged)'].apply(pd.to_numeric).describe()

In [14]:
df_lower_currency_removed = df_lower_currency3_removed

In [15]:
## View new DataFrame
#df_lower_currency_removed

In [16]:
## Creating Separate DataFrames for each Journal Title

plos_one_isolated=df_lower_currency_removed.loc[lambda df: df['Journal title'] == 'plos one', :]

jbc_isolated=df_lower_currency_removed.loc[lambda df: df['Journal title'] == 'journal of biological chemistry', :]

neuroimage_isolated=df_lower_currency_removed.loc[lambda df: df['Journal title'] == 'neuroimage', :]

plos_pathogens_isolated=df_lower_currency_removed.loc[lambda df: df['Journal title'] == 'plos pathogens', :]

plos_genetics_isolated=df_lower_currency_removed.loc[lambda df: df['Journal title'] == 'plos genetics', :]

In [17]:
## Calculating Mean, Median, Standard deviation for PLOS One article Costs

## Describe PLOS One costs
plos_one_isolated['COST (£) charged to Wellcome (inc VAT when charged)'].apply(pd.to_numeric).describe()

count       190.000000
mean       1983.461263
std       13908.126819
min         122.310000
25%         852.455000
50%         898.475000
75%        1044.737500
max      192645.000000
Name: COST (£) charged to Wellcome (inc VAT when charged), dtype: float64

In [18]:
## Remove Extreme Costs 

r = plos_one_isolated['COST (£) charged to Wellcome (inc VAT when charged)'].apply(pd.to_numeric).quantile(0.99)

plos_one_isolated_reduced = plos_one_isolated[plos_one_isolated['COST (£) charged to Wellcome (inc VAT when charged)'].apply(pd.to_numeric) < r]

In [19]:
## Describe Article Costs

In [20]:
plos_one_isolated_reduced['COST (£) charged to Wellcome (inc VAT when charged)'].apply(pd.to_numeric).describe()

count     182.000000
mean      934.655824
std       201.847589
min       122.310000
25%       850.125000
50%       896.975000
75%      1040.737500
max      1785.360000
Name: COST (£) charged to Wellcome (inc VAT when charged), dtype: float64

**PLOS One Article Cost Statistics <br>
mean:      934.655824 <br>
median:    896.975000 <br>
std:       201.847589**


In [21]:
jbc_isolated['COST (£) charged to Wellcome (inc VAT when charged)'].apply(pd.to_numeric).describe()

count      53.000000
mean     1434.747925
std       415.984485
min       265.670000
25%      1152.720000
50%      1314.530000
75%      1610.310000
max      2501.070000
Name: COST (£) charged to Wellcome (inc VAT when charged), dtype: float64

**Journal of Biological Chemistry Cost Statistics <br>
mean:     1434.747925 <br>
median:      1314.530000 <br>
std:       415.984485**

In [22]:
neuroimage_isolated['COST (£) charged to Wellcome (inc VAT when charged)'].apply(pd.to_numeric).describe()

count      29.000000
mean     2215.168276
std       266.653947
min      1747.160000
25%      2100.540000
50%      2326.430000
75%      2396.570000
max      2503.340000
Name: COST (£) charged to Wellcome (inc VAT when charged), dtype: float64

**Neuroimage Cost Statistics <br>
mean:     2215.168276 <br>
median:      2326.430000 <br>
std:       266.653947**

In [23]:
plos_pathogens_isolated['COST (£) charged to Wellcome (inc VAT when charged)'].apply(pd.to_numeric).describe()

count      24.000000
mean     1609.714583
std       198.700369
min      1254.020000
25%      1440.000000
50%      1600.520000
75%      1760.310000
max      2015.040000
Name: COST (£) charged to Wellcome (inc VAT when charged), dtype: float64

**PLOS Pathogens Cost Statistics <br>
mean:     1609.714583 <br>
median:      1600.520000 <br>
std:       198.700369**

In [24]:
plos_genetics_isolated['COST (£) charged to Wellcome (inc VAT when charged)'].apply(pd.to_numeric).describe()

count      24.000000
mean     1674.105000
std       180.284357
min      1394.050000
25%      1484.862500
50%      1718.390000
75%      1778.225000
max      2015.040000
Name: COST (£) charged to Wellcome (inc VAT when charged), dtype: float64

**PLOS Genetics Cost Statistics <br>
mean:     1674.105000 <br>
median:      1718.390000 <br>
std:       180.284357**